servicetalk-http-api/src/main/java/io/servicetalk/http/api/ProxyConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.transport.api.ClientSslConfig;
+
+import java.util.function.Consumer;
+
+/**
+ * Configuration for a proxy.
+ *
+ * @param <A> the type of address
+ */
+public interface ProxyConfig<A> {
+
+    /**
+     * Address of the proxy.
+     * <p>
+     * Usually, this is an unresolved proxy address and its type must match the type of address before resolution used
+     * by {@link SingleAddressHttpClientBuilder}. However, if the client builder was created for a resolved address,
+     * this address must also be already resolved. Otherwise, a runtime exceptions will occur.
+     *
+     * @return address of the proxy
+     */
+    A address();
+
+    /**
+     * An initializer for {@link HttpHeaders} related to
+     * <a href="https://datatracker.ietf.org/doc/html/rfc9110#section-9.3.6">HTTP/1.1 CONNECT</a> request.
+     * <p>
+     * When this {@link ProxyConfig} is used for secure proxy tunnels (when
+     * {@link SingleAddressHttpClientBuilder#sslConfig(ClientSslConfig) ClientSslConfig} is configured) the tunnel is
+     * always initialized using {@code HTTP/1.1 CONNECT} request. This {@link Consumer} can be used to set custom
+     * {@link HttpHeaders} for that request, like {@link HttpHeaderNames#PROXY_AUTHORIZATION proxy-authorization},
+     * tracing, or any other header.
+     *
```

Comment:
If (are we?) doing any retries on the connect request, this comment I think should clarified if it is called N times or just once.

Reply:
No, there are no retries specifically for a `CONNECT` request. In case of a general retry, it will go through process of opening a new connection and then it's not different from any other new connection event, regardless of if it was retried or not.